lib/remote-model.js

```diff
@@ -1,120 +1,96 @@
+        this.request(config, callback);
+    }
+
+    save(callback) {
         this.prepare((err, data) => {
 
-            if (err) {
-                return callback(err);
-            }
+            if (err) { return callback(err); }
 
             data = JSON.stringify(data);
 
-            let reqConf = this.requestConfig(options);
-
-            reqConf.method = options.method || 'POST';
+            let config = this.requestConfig({
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json',
+                    'Content-Length': Buffer.byteLength(data)
+                }
+            });
 
-            reqConf.headers = _.extend({
-                'Content-Type': 'application/json',
-                'Content-Length': Buffer.byteLength(data)
-            }, reqConf.headers || {});
+            this.request(config, data, callback);
 
-            this.request(reqConf, data, callback);
         });
     }
 
-    delete(options, callback) {
-        if (typeof options === 'function' && arguments.length === 1) {
-            callback = options;
-            options = {};
-        } else if (!options) {
-            options = {};
-        }
-        let reqConf = this.requestConfig(options);
-        reqConf.method = options.method || 'DELETE';
-        this.request(reqConf, callback);
+    delete(callback) {
+        let config  = this.requestConfig({method: 'DELETE'});
+        this.request(config, callback);
     }
 
-    requestConfig(options) {
-        let reqConf = this.url(options);
-        if (typeof reqConf === 'string') {
-            reqConf = url.parse(reqConf);
+    requestConfig(config) {
+        config.uri = this.url();
+
+        let auth = this.auth();
+
+        if (auth) {
+            config.auth = auth;
         }
-        return reqConf;
+
+        if (config.headers) {
+            config.headers = _.extend({}, config.headers);
+        }
+
+        return config;
     }
 
-    request(originalSettings, body, callback) {
+    request(settings, body, callback) {
         if (typeof body === 'function' && arguments.length === 2) {
             callback = body;
             body = undefined;
         }
 
-        let settings = _.extend({}, originalSettings);
-        settings.timeout = settings.timeout || this.options.timeout;
-        settings.uri = settings.uri || settings.url || url.format(settings);
-        settings.body = settings.body || body || settings.data;
+        let requestSettings = _.clone(settings);
 
-        settings = _.omit(settings, util.urlKeys, 'data', 'url');
-
-        settings.auth = this.auth();
-        if (typeof settings.auth === 'string') {
-            let auth = settings.auth.split(':');
-            settings.auth = {
-                user: auth.shift(),
-                pass: auth.join(':'),
-                sendImmediately: true
-            };
-        }
+        requestSettings.body = body;
```

Comment:
for other reviewers: this is cloned and set so that the post body is not kept around in memory while the settings is kept around.